rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -6551,6 +6552,312 @@ private void assertRefundInProcessPegInVersionLegacy(
+            .withBtcBlockStoreFactory(mockFactory)
+            .withExecutionBlock(executionBlock)
+            .withRepository(repository)
+            .build();
+
+        Keccak256 derivationArgumentsHash = PegTestUtils.createHash3(0);
+        Keccak256 fastBridgeDerivationHash = bridgeSupport.getFastBridgeDerivationHash(
+            derivationArgumentsHash,
+            userRefundBtcAddress,
+            lpBtcAddress,
+            lbcAddress
+        );
+
+        if (includeActiveFederation) {
+            Address activeFederationAddress = getFastBridgeAddressFromRedeemScript(
+                activeFederation.getRedeemScript(),
+                Sha256Hash.wrap(fastBridgeDerivationHash.getBytes())
+            );
+            tx.addOutput(valueToSend, activeFederationAddress);
+        }
+
+        if (includeRetiringFederation) {
+            Address retiringFederationAddress = getFastBridgeAddressFromRedeemScript(
+                retiringFederation.getRedeemScript(),
+                Sha256Hash.wrap(fastBridgeDerivationHash.getBytes())
+            );
+            tx.addOutput(valueToSend, retiringFederationAddress);
+        }
+        tx.addInput(
+            Sha256Hash.wrap(fastBridgeDerivationHash.getBytes()),
+            0, ScriptBuilder.createInputScript(null, srcKey));
+
+        List<Sha256Hash> hashes = new ArrayList<>();
+        hashes.add(tx.getHash());
+        PartialMerkleTree pmt = new PartialMerkleTree(bridgeConstants.getBtcParams(), bits, hashes, 1);
+        Sha256Hash merkleRoot = pmt.getTxnHashAndMerkleRoot(new ArrayList<>());
+        co.rsk.bitcoinj.core.BtcBlock registerHeader = new co.rsk.bitcoinj.core.BtcBlock(
+            bridgeConstants.getBtcParams(),
+            1,
+            PegTestUtils.createHash(2),
+            merkleRoot,
+            1,
+            1,
+            1,
+            new ArrayList<>()
+        );
+        // simulate blockchain
+        mockChainOfStoredBlocks(
+            btcBlockStore,
+            registerHeader,
+            height + bridgeConstants.getBtc2RskMinimumAcceptableConfirmations(),
+            height
+        );
+
+        InternalTransaction rskTx = new InternalTransaction(
+            Keccak256.ZERO_HASH.getBytes(),
+            0,
+            0,
+            null,
+            null,
+            null,
+            lbcAddress.getBytes(),
+            null,
+            null,
+            null,
+            null
+        );
+
+        co.rsk.core.Coin preCallLbcAddressBalance = repository.getBalance(lbcAddress);
+
+        BigInteger result = bridgeSupport.registerFastBridgeBtcTransaction(
+            rskTx,
+            tx.bitcoinSerialize(),
+            height,
+            pmt.bitcoinSerialize(),
+            derivationArgumentsHash,
+            userRefundBtcAddress,
+            lbcAddress,
+            lpBtcAddress,
+            true
+        );
+
+        if (includeActiveFederation && !includeRetiringFederation) {
```

Comment:
Why not assert when retiring federation is not included?

Reply:
It could be included. Just need to modify the code  a bit to get it working for when there is an output to the retiring federation. Actually, I fixed this in one of the later tasks.